doc/sf/number_series.qbk

```diff
@@ -259,9 +259,64 @@ Passing a value greater than `max_prime` results in a __domain_error being raise
 
 This function is `constexpr` only if the compiler supports C++14 constexpr functions.
 
-[endsect] [/section:primes]
+[endsect] [/section:primes Prime Numbers]
 
-[endsect] [/Number Series]
+[section:prime_sieve Prime Sieve]
+
+[h4 Synopsis]
+
+``
+#include <boost/math/special_functions/prime_sieve.hpp>
+``
+
+namespace boost { namespace math {
+
+    template<class ExecutionPolicy, typename Z, class OutputIterator>
+    auto prime_sieve(ExecutionPolicy&& policy, Z upper_bound, OutputIterator output) -> decltype(output)
+
+    template<class ExecutionPolicy, class Z, class OutputIterator>
+    auto prime_range(ExecutionPolicy&& policy, Z lower_bound, Z upper_bound, OutputIterator output) -> decltype(output)
+
+    template<typename Z, class OutputIterator>
+    auto prime_sieve(Z upper_bound, OutputIterator output) -> decltype(output)
+
+    template<class Z, class OutputIterator>
+    auto prime_range(Z lower_bound, Z upper_bound, OutputIterator output) -> decltype(output)
+
+
+}} // End namespaces
+
+[h4 Description]
+
+There are two sets of functions available `prime_sieve` and `prime_range`. `prime_sieve` will return all primes in the
+range 2 to `upper_bound` inclusive. `prime_range` will return all the primes in the range `lower_bound` to `upper_bound`
+inclusive.
```

Comment:
Generally we don't do inclusive in C++; what is the motivation to include `upper_bound`?

Reply:
It was a design choice. Wolfram-alpha uses the range `[lower_bound, upper_bound]` while sympy uses `[lower_bound, upper_bound)`. It made sense to me to include the upper bound, but it should rarely make a difference.

Reply:
I think for consistency with the C++ std lib, I would go exclusive and use `[lower, upper)` as the bounds.